packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -281,6 +281,15 @@ export default function(
           'Did you mean componentWillUnmount()?',
         name,
       );
+      const noComponentDidReceiveProps =
+        typeof instance.componentDidReceiveProps !== 'function';
+      warning(
+        noComponentDidReceiveProps,
+        '%s has a method called ' +
+          'componentDidReceiveProps(). But there is no such lifecycle method. ' +
+          'Did you mean componentDidUpdate()?',
```

Comment:
Seems more likely they mean `componentWillReceiveProps()`? Let's include both

Reply:
I think in most cases people *would* mean `componentWillReceiveProps`.

Maybe we should say

>If you meant to update the state in response to changing props, use `componentWillReceiveProps()`. If you meant to perform manual DOM manipulations or fetch data, use `componentDidUpdate()`.

Reply:
I like that, though will also be seen for non-DOM users.

Reply:
Maybe

>If you meant to update the state in response to changing props, use `componentWillReceiveProps()`. If you meant to fetch data or run some code after React has updated the UI, use `componentDidUpdate()`.

Reply:
"some code" -> "side-effects or mutations"

?

Reply:
👍